js/pulse.js

```diff
@@ -363,7 +364,9 @@
 
                 this.ready = true;
                 this._deviceContainer = adManagerMetadata.pulse_device_container;
-                this._persistentId = adManagerMetadata.pulse_persistent_id;
+                if (adManagerMetadata.pulse_persistent_id) {
+                  this._persistentId = adManagerMetadata.pulse_persistent_id;
```

Comment:
We want to use the device id being used by the player as the persistent id, except when the pulse_persistent_id is already defined in the metadata. In that case we will allow overriding the player device id, this is to maintain backward compatibility with the way it currently works.

DEVICE_ID_SET is triggered when the player identifies the device id from the local storage. This device id has been around for many years and used in identifying devices for video analytics. We want to use the same device id for ads.
The corresponding changes to the core player are in this PR that I have added you as a reviewer to https://git.corp.ooyala.com/projects/IO/repos/mjolnir/pull-requests/1/diff

DEVICE_ID_SET is triggered on loading the player therefore it gets triggered before loadMetadata